### PR TITLE
Fix DbKvs.getRangeOfTimestamps() only returning the first page

### PR DESCRIPTION
**Pull request by gbonik**
The expectation is to return all results that fall into the range
regardless of the page size. This is how all other KVSs work
at least, and also consistent with getRange().

I'm planning to delete all of this code soon, and just need it to "work" in order
to execute the intermediate steps of my big sweep refactor/rewrite.

**Goals (and why)**:
Fix broken stuff that is blocking my work
**Implementation Description (bullets)**:
Load an extra row. Add a test.
**Concerns (what feedback would you like?)**:
It's still wrong, but that's okay (read above)
**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1872)

### CODE REVIEW

**Comment by gbonik** [comment]
This fix doesn't actually work, please don't merge this yet.

**Comment by gbonik** [comment]
Pushed a proper fix, changed the test, and updated the PR description.

**Comment by jeremyk-91** [comment]
Right, I was initially a little puzzled as to how adding 1 to the query sizes worked. I'll start reviewing this now.

**Comment by gbonik** [comment]
@jeremyk-91 The logic to determine whether we reached the end was to to check if there was a remaining result from the SQL query, so my idea was to load that extra result. A pretty common trick for implementing paging. It didn't work because "range.getBatchHint()" and the "batchSize" are two different numbers.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java [hunk 1]
```diff
@@ -123,6 +127,13 @@ private TimestampsByCellResultWithToken checkNextEntryAndCreateToken() {
             }
         } else {
             flushRowBuffer();
+            if (currentRow != null) {
+                byte[] nextRow = RangeRequests.getNextStartRowUnlessTerminal(reverse, currentRow);
+                if (nextRow != null) {
+                    moreResults = true;
+                    token = ImmutableToken.builder().row(nextRow).shouldSkip(false).build();
+                }
+            }
```
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java]: We could extract this to `buildTokenIfMoreRowsAvailable` or something along those lines.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java [hunk 2]
```diff
@@ -1079,6 +1079,23 @@ private void testGetRangeWithTimestamps(boolean reverse) {
...
+                Cell.create(row1, column0), value0_t0,
+                Cell.create(row2, column0), value0_t0),
+            TEST_TIMESTAMP);
+        RangeRequest range = RangeRequest.all().withBatchHint(1);
+        List<RowResult<Set<Long>>> results = ImmutableList.copyOf(
+                keyValueService.getRangeOfTimestamps(TEST_TABLE, range, TEST_TIMESTAMP + 1));
+        assertEquals(3, results.size());
+        assertTrue(Arrays.equals(row0, results.get(0).getRowName()));
+        assertTrue(Arrays.equals(row1, results.get(1).getRowName()));
+        assertTrue(Arrays.equals(row2, results.get(2).getRowName()));
```
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: For these, I'd prefer we use `assertArrayEquals(expected, actual)` instead. We'll get a more helpful assertion failure message if something breaks here.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java [hunk 3]
```diff
@@ -1079,6 +1079,23 @@ private void testGetRangeWithTimestamps(boolean reverse) {
     }
 
     @Test
+    public void testGetRangeOfTimestampsReturnsAllRows() {
+        keyValueService.put(TEST_TABLE,
+            ImmutableMap.of(
+                Cell.create(row0, column0), value0_t0,
+                Cell.create(row1, column0), value0_t0,
+                Cell.create(row2, column0), value0_t0),
+            TEST_TIMESTAMP);
```
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: Ah, the above needs to be indented 4 more spaces in b/c of checkstyle.

```
[00:02:37] [ant:checkstyle] [ERROR] /home/ubuntu/atlasdb-internal-testing/atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java:1084: 'ImmutableMap' have incorrect indentation level 12, expected level should be 16. [Indentation]
[00:02:37] [ant:checkstyle] [ERROR] /home/ubuntu/atlasdb-internal-testing/atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java:1088: 'TEST_TIMESTAMP' have incorrect indentation level 12, expected level should be 16. [Indentation]
```

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java (hunk 1 above)
Comment (gbonik) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java]: We could but I'm planning to delete all of this code soon.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java (hunk 3 above)
Comment (gbonik) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: Are you guys using the "baseline" checkstyle? I feel like at least half of the rules there don't make much sense.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java (hunk 2 above)
Comment (gbonik) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: Yep, forgot that it existed. Done.

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java (hunk 1 above)
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/TimestampsByCellResultWithToken.java]: Sure, no problem.

File: atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java (hunk 3 above)
Comment (jeremyk-91) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: For the most part I think so, though we do exclude a couple of rules - @gsheasby would probably know this better!

Comment (gsheasby) [inline atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueServiceTest.java]: @gbonik yep, we recently upgraded to Baseline 0.13.0.

**Pull request merged**